### ISSUE TITLE
Protozfitsreader crashes when zfits file does not contain Trigger Output Traces

### ISSUE THREAD

**Issue by calispac** [cta-sst-1m/protozfitsreader#26]
I was trying to read a recent `zfits` where the Trigger Output Traces where not recorded.
By trying to read it with `v. 0-44.1` I got the following error : 

```
digicam-view /sst1m/raw/2018/02/21/SST1M01/SST1M01_0_000.000.fits.fz 
Traceback (most recent call last):
  File "/home/alispach/.anaconda3/envs/digicampipe/bin/digicam-view", line 11, in <module>
    load_entry_point('digicampipe', 'console_scripts', 'digicam-view')()
  File "/home/alispach/Documents/PhD/ctasoft/digicampipe/digicampipe/scripts/digicamview.py", line 15, in entry
    display.draw()
  File "/home/alispach/Documents/PhD/ctasoft/digicampipe/digicampipe/visualization/__init__.py", line 226, in draw
    self.next()
  File "/home/alispach/Documents/PhD/ctasoft/digicampipe/digicampipe/visualization/__init__.py", line 157, in next
    for i, event in zip(range(step), self.event_stream):
  File "/home/alispach/Documents/PhD/ctasoft/digicampipe/digicampipe/io/event_stream.py", line 41, in event_stream
    for event in data_stream:
  File "/home/alispach/Documents/PhD/ctasoft/digicampipe/digicampipe/io/zfits.py", line 76, in zfits_event_source
    for event_counter, event in enumerate(ZFile(url)):
  File "/home/alispach/.anaconda3/envs/digicampipe/lib/python3.5/site-packages/protozfitsreader/__init__.py", line 181, in __next__
    return Event(event, self.run_id)
  File "/home/alispach/.anaconda3/envs/digicampipe/lib/python3.5/site-packages/protozfitsreader/__init__.py", line 219, in __init__
    _e.trigger_output_patch7)
  File "/home/alispach/.anaconda3/envs/digicampipe/lib/python3.5/site-packages/protozfitsreader/__init__.py", line 266, in _prepare_trigger_output
    _a = toNumPyArray(_a)
  File "/home/alispach/.anaconda3/envs/digicampipe/lib/python3.5/site-packages/protozfitsreader/__init__.py", line 302, in toNumPyArray
    any_array_type_cannot_convert_exception_text[a.type])
ValueError: ('Conversion to NumpyArray failed with error:\n%s', 'This any array has no defined type')
```

By looking into the file using this command : 

```
[ctauser@pdp CamerasToACTL]$ Build.ACS/bin/zfitsdump ~/data/raw/2018/02/21/SST1M01/SST1M01_0_000.000.fits.fz --num_rows 0
#################################### INPUT #####################################
File : /home/ctauser/data/raw/2018/02/21/SST1M01/SST1M01_0_000.000.fits.fz
Table: Events
################################# FITS HEADER ##################################
DATE     2018-02-21T12:58:41
DATEEND  2018-02-21T13:00:19
ORIGIN   CTA
WORKPKG  ACTL
#################################### DATA ######################################
Table contains 290 messages of type DataModel.CameraEvent. Columns:
telescopeID(INT32)
eventNumber(UINT32)
hiGain_waveforms_samples(int16[64800])
hiGain_waveforms_pixelsIndices(uint16[1296])
hiGain_waveforms_baselines(int16[1296])
local_time_nanosec(UINT64)
local_time_sec(UINT64)
pixels_flags(uint16[1296])
event_type(UINT32)
################################################################################
```

 @Etienne12345 and I found that the file does not contain trigger output traces. I think we should avoid that the protozfitsreader crashes if it is missing the trigger output/input traces

**Comment by dneise** [comment]
Is it really the protozfits reader which crashes? Or is it the python "wrapper" around the protozfits reader which crashes? (Which would make it my fault not Etiennes).

**Comment by dneise** [comment]
>   File "/home/alispach/.anaconda3/envs/digicampipe/lib/python3.5/site-packages/protozfitsreader/__init__.py", line 266, in _prepare_trigger_output
    _a = toNumPyArray(_a)
  File "/home/alispach/.anaconda3/envs/digicampipe/lib/python3.5/site-packages/protozfitsreader/__init__.py", line 302, in toNumPyArray
    any_array_type_cannot_convert_exception_text[a.type])

It is may fault.

**Comment by dneise** [comment]
So how do we want to treat this case? 

We can catch this exception and return an array full of nans, which might or might not cause an exception in later code, which is not fit to work with nans. 

Or it does not throw an exception at all, but the user is puzzled because the result of some analysis returns only nans .. and the user has no idea why...

Or we stick to the current behaviour, i.e. we say: "A valid SST1M digicam file must have this field, otherwise we consider the file simply as 'broken'".

I would like repeat that code, which tries to solve all corner cases which might happen, or might have happened once in the past of digicam, will grow to some enormous blob of edge-case-treatment. Making it hard to see, what is actually happening. So if this was a mistake during file creation, then ... maybe this file is simply broken and throwing an exception is *exactly the right thing* to do for our reader.

**Comment by calispac** [comment]
I think this particular field is not always required for data analysis. The data taker could just ignore it if he/she does not want to overload the disk. 

So I would prefer that it sends nan values + a warning: 
"The file %s does not contain the field %s, will use nan values !"

**Comment by dneise** [comment]
I "fixed" this in a hotfix relative to v0.44.1 

https://github.com/cta-sst-1m/protozfitsreader/commit/e264d3ddb03655f0496cd0604e77b40aabb7b367

But be aware that the master is now at v0.44.3 maybe digicampipe wants to use v0.44.3 one day?

**Comment by calispac** [comment]
Thanks I checked it works !
